templates/mesh-gateway-deployment.yaml

```diff
@@ -1,7 +1,7 @@
 {{- if .Values.meshGateway.enabled }}
 {{- if not .Values.connectInject.enabled }}{{ fail "connectInject.enabled must be true" }}{{ end -}}
 {{- if not .Values.client.grpc }}{{ fail "client.grpc must be true" }}{{ end -}}
-{{- if and .Values.global.bootstrapACLs (ne .Values.meshGateway.consulServiceName "") (ne .Values.meshGateway.consulServiceName "mesh-gateway") }}{{ fail "if global.bootstrapACLs is true, meshGateway.consulServiceName cannot be set" }}{{ end -}}
+{{- if and (or .Values.global.acls.manageSystemACLs .Values.global.bootstrapACLs) (ne .Values.meshGateway.consulServiceName "") (ne .Values.meshGateway.consulServiceName "mesh-gateway") }}{{ fail "if global.acls.manageSystemACLs is true, meshGateway.consulServiceName cannot be set" }}{{ end -}}
```

Comment:
This error message won't help someone track down what's going wrong if they're using `bootstrapACLs`. Though, this is likely an error someone would hit when first setting things up, and then they should be using the other flag. So it's probably okay.

Reply:
Yeah I thought putting verbiage about how it could be triggered by either bootstrapACLs or manageSystemACLs would make it more confusing. I'm hoping they'd be able to trace it through since they'd be like "I'm not setting manageSystemACLs", look at that flag and realize that bootstrapACLs was deprecated.